src/foam/dao/AbstractJDAO.java

```diff
@@ -283,47 +277,24 @@ public void removeAll_(final X x, long skip, final long limit, Comparator order,
     getDelegate().removeAll_(x, skip, limit, order, predicate);
   }
 
-  protected FObject difference(FObject o, FObject n) {
-    FObject diff = o.hardDiff(n);
-    // no difference, then return null
-    if ( diff == null ) return null;
-    // get the PropertyInfo for the id
-    PropertyInfo idInfo = (PropertyInfo) getOf().getAxiomByName("id");
-    // set id property to new instance
-    idInfo.set(diff, idInfo.get(o));
-    return diff;
-  }
-
-  protected FObject mergeChange(FObject o, FObject c) {
-    //if no change to merge, return FObject;
-    if ( c == null ) return o;
-    //merge change
-    return maybeMerge(o, c);
-  }
-
-  protected FObject maybeMerge(FObject o, FObject c) {
-    if ( o == null ) return o = c;
+  //add diff property to old property
+  protected FObject mergeFObject(FObject oldFObject, FObject diffFObject) {
+    if ( diffFObject == null ) return oldFObject;
```

Comment:
When is diffFObject null? If it's null couldn't I just not call mergeFObject() and not call put()?